mbuild/tests/test_polymer.py

```diff
@@ -32,3 +33,50 @@ def test_block_copolymer(self, ch2, ester):
         assert n_elements['H'] == n * len(sequence)
         assert n_elements['O'] == n * len(sequence)
         assert abba.n_bonds == n * 2 * len(sequence) + (n * len(sequence) - 1)
+
+    def test_polymer_caps(self, ch2, ch3, ester):
+        n = 6
+        c6 = mb.Polymer(ch2, n=n,caps=(ch3,ester),cap_ports=('up','down'))
+        assert c6.n_particles == n * 3 + 7
+        assert c6.n_bonds == n * 2 + (n - 1) + 7
+        assert c6.children[6].name == 'CH3'
+        assert c6.children[7].name == 'Ester'
+        assert len(c6.all_ports()) == 1
+        assert assert_port_exists('up', c6.children[7])
```

Comment:
This is somewhat of a niche case, but I think it would make more sense if the `'up'` port here was accessible from `c6` rather than `c6.children[7]`. Maybe inside the `Polymer` class we add labels to `self` for any ports still available on the end caps? Although I suppose this could get messy if there were multiple ports with the same label...